src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -1054,6 +1070,10 @@ private void BlockDownloaded(uint256 blockHash, Block block, int peerId)
                     return;
                 }
             }
+            else
+            {
+                this.logger.LogDebug("Block '{0}' failed to be delivered.", blockHash);
```

Comment:
I would actually suggest we throw an exception here, this can not happen anymore right?

Reply:
It can. See my msg above.

Reply:
hmmm so basically a failed download is a valid scenario even if we have a peer that can provide this block?

Reply:
Yes, but only due to the race condition.
Peers disconnected, a new peer connected that can provide that block, now fail callback executed.